src/Sarif/Core/Artifact.cs

```diff
@@ -88,13 +88,21 @@ private static ArtifactContent GetEncodedFileContents(IFileSystem fileSystem, st
             var fileContent = new ArtifactContent();
             byte[] fileContents = fileSystem.ReadAllBytes(filePath);
 
-            if (SarifWriters.MimeType.IsBinaryMimeType(mimeType) || inputFileEncoding == null)
+            if (SarifWriters.MimeType.IsBinaryMimeType(mimeType))
             {
                 fileContent.Binary = Convert.ToBase64String(fileContents);
             }
             else
             {
-                fileContent.Text = inputFileEncoding.GetString(fileContents);
+                inputFileEncoding ??= new UTF8Encoding();
+                try
+                {
+                    fileContent.Text = inputFileEncoding.GetString(fileContents);
+                }
+                catch (Exception)
```

Comment:
>Exception [](start = 23, length = 9)

I'd removed this exception handler. First, I don't think it's likely that Encoding.GetString will fail against C# string values. Do you know of an expected negative condition? If there *is* one we need to handle, we should catch a more precise exception (swallowing all Exceptions is not a good idea).

Reply:
We can receive a few: [https://docs.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.getstring?view=netcore-3.1](https://docs.microsoft.com/en-us/dotnet/api/system.text.utf8encoding.getstring?view=netcore-3.1), but normally we wouldn't face that. I will remove that for now. Thanks for reviewing!

---
In reply to: [513607954](https://github.com/microsoft/sarif-sdk/pull/2134#discussion_r513607954) [](ancestors = 513607954)